### PR TITLE
burp 2.0.44 (devel)

### PR DESCRIPTION
**Pull request by ilovezfs**
- add devel spec and use same steps for head spec

### CODE REVIEW

**Comment by ilovezfs** [comment]
CC @bfontaine RE: https://github.com/Homebrew/legacy-homebrew/pull/40393

File: Formula/burp.rb [hunk 1]
```diff
@@ -13,14 +17,51 @@ class Burp < Formula
...
-  # files
-  patch :DATA
-
   def install
+    if build.devel? || build.head?
+      resource("uthash").stage do
+        cd "libut" do
+          system "make"
+          (buildpath/"uthash/lib").install "libut.a"
+        end
```
Comment (bfontaine) [inline Formula/burp.rb]: That’s probably nitpicking but you can do this without `cd`:

```ruby
        system "make", "-C", "libut"
        (buildpath/"uthash/lib").install "libut/libut.a"
```

Comment (ilovezfs) [inline Formula/burp.rb]: Yeah it's deliberate, since I'm not a big fan of "-C" for git or make :)

File: Formula/burp.rb [hunk 2]
```diff
@@ -13,14 +17,51 @@ class Burp < Formula
...
+
   depends_on "librsync"
   depends_on "openssl"
 
-  # patches to change directories to brew conventions in Makefile and config
-  # files
-  patch :DATA
-
   def install
+    if build.devel? || build.head?
```
Comment (DomT4) [inline Formula/burp.rb]: I'd just feel free to use `unless build.stable?` here, if you wanted to. Where it applies to both devel & head `unless build.stable?` is a valid syntax choice.

**Comment by ilovezfs** [comment]
@DomT4 @bfontaine PR refreshed.

**Comment by DomT4** [comment]
Test is failing. Related to the moved executable apparently.

**Comment by ilovezfs** [comment]
LOL yeah, stable decided that it should put them in `/usr/local/burp/1.4.40_1/usr/sbin/`.

**Comment by DomT4** [comment]
Wow 😓.

**Comment by ilovezfs** [comment]
I guess I'll do this
```
diff --git a/Formula/burp.rb b/Formula/burp.rb
index b8a594c..08c8409 100644
--- a/Formula/burp.rb
+++ b/Formula/burp.rb
@@ -61,9 +61,14 @@ class Burp < Formula
       system "autoreconf", "-fiv"
     end
 
-    system "./configure", "--prefix=#{prefix}",
-                          "--sysconfdir=#{etc}/burp",
-                          "--localstatedir=#{var}/burp"
+    args = %W[
+      --prefix=#{prefix}
+      --sysconfdir=#{etc}/burp
+      --localstatedir=#{var}/burp
+    ]
+    args << "--sbindir=#{sbin}" if build.stable?
+
+    system "./configure", *args
     system "make", "install"
   end
```

Or could just do `--sbindir=#{sbin}` for both stable and devel, even though devel gets it right out of the box.

**Comment by DomT4** [comment]
Proposed diff seems alright 👍.

**Comment by ilovezfs** [comment]
:green_apple:

**Comment by DomT4** [comment]
Is all the stuff we've had to patch fixes in the `devel`, etc releases?

**Comment by ilovezfs** [comment]
The patch applies cleanly because I think all it's doing is switching up installation locations as opposed to actually fixing bugs.

**Comment by ilovezfs** [comment]
Ah nevermind, I see I moved :DATA into stable... I'll investigate further.

**Comment by ilovezfs** [comment]
devel has
```
        $(AM_V_at)$(MKDIR_P) $(DESTDIR)$(localstatedir)/spool/burp
        $(AM_V_at)$(MKDIR_P) $(DESTDIR)$(runstatedir)
./configs/client/burp.conf.in:pidfile = @runstatedir@/burp.client.pid
./configs/server/burp.conf.in:pidfile = @runstatedir@/burp.server.pid
clientconfdir
clientconfdir = $(sysconfdir)/clientconfdir
./configs/client/burp.conf.in:ca_burp_ca = @sbindir@/burp_ca
./configs/server/burp.conf.in:ca_burp_ca = @sbindir@/burp_ca
./configs/client/burp.conf.in:ca_csr_dir = @sysconfdir@/CA-client
./configs/client/burp.conf.in:ssl_cert_ca = @sysconfdir@/ssl_cert_ca.pem
./configs/server/burp.conf.in:ssl_cert_ca = @sysconfdir@/ssl_cert_ca.pem
./configs/client/burp.conf.in:ssl_key = @sysconfdir@/ssl_cert-client.key
```
So all is "properly" prefixed now instead of hard coded.

**Comment by ilovezfs** [comment]
@DomT4 :ship:?

**Comment by DomT4** [comment]
You may possibly want to create `#{var}/burp` in `postinstall`, but if not the case 👍.

**Comment by ilovezfs** [comment]
@DomT4 PR refreshed. Found a couple of bugs. Upstream changed `make install` to not include `install-configs`. So you can either do `make install ; make install-configs` or `make install-all` which does both. I opted for the latter in case they move anything else into that umbrella. The other bug is that in stable, even though the configure sets var to var/burp, if you actually look at what the patch does, it ends up using simply var not var/burp. So stable has been doing this:
```
/usr/local/var
/usr/local/var/run
/usr/local/var/spool
/usr/local/var/spool/burp
```
Do we prefer that or what stable was supposedly doing according to configure (but wasn't) and what devel was indeed doing:
```
/usr/local/var
/usr/local/var/burp
/usr/local/var/burp/run
/usr/local/var/burp/spool
/usr/local/burp/spool/burp
```
If the latter is preferred, then the patch for stable can be fixed accordingly.

Either way, it doesn't need to be part of post install because the bottles include them in the `.bottle` directory, which then gets installed into the prefix. So they get installed by the bottle because they got created during the build.

File: Formula/burp.rb [hunk 3]
```diff
@@ -13,19 +17,60 @@ class Burp < Formula
...
+      system "make", "install"
+    else
+      resource("uthash").stage do
+        system "make", "-C", "libut"
+        (buildpath/"uthash/lib").install "libut/libut.a"
+        (buildpath/"uthash/include").install Dir["src/*"]
+      end
+
+      ENV.prepend "CFLAGS", "-I#{buildpath}/uthash/include"
+      ENV.prepend "CXXFLAGS", ENV.cflags
```
Comment (ilovezfs) [inline Formula/burp.rb]: Hmm I wonder if this should be just CPPFLAGS instead of CFLAGS and CXXFLAGS separately since it's setting an include.

**Comment by DomT4** [comment]
> If the latter is preferred, then the patch for stable can be fixed accordingly.

Either are fine really, but if you want to enforce consistency between stable/devel here I've no objections. All the `var` paths should be created in `post_install` as well, ideally.

**Comment by ilovezfs** [comment]
@DomT4 PR refreshed.

**Comment by DomT4** [comment]
Not a massive fan of the lack of spacing between `sha256` & the next line in `stable do`, whether that's a patch or deps, but I seem to be losing that argument in Homebrew so 👍.

**Comment by ilovezfs** [comment]
>Not a massive fan of the lack of spacing between sha256 & the next line in stable do, whether that's a patch or deps, but I seem to be losing that argument in Homebrew so :+1:

I agree with you, but I thought you'd already said that was a lost cause so this was the "correct" way now hehe.

**Comment by DomT4** [comment]
Well, between us two, I'm pretty sure we have more commits to `core` this year than everyone else combined.. so if we wanted to change it I suspect we could 😆.

**Comment by ilovezfs** [comment]
>... so if we wanted to ...

Shhh someone might here our plotting.

**Comment by ilovezfs** [comment]
@DomT4 fixed the grotesque lack of spacing in post: https://github.com/Homebrew/homebrew-core/commit/6ee197091fa7f09232c95556e319679b3b99e6c8#diff-4fb9179650da103a71e3042ee5f3221fR9

**Comment by DomT4** [comment]
> Shhh someone might hear our plotting.

I think at this point the coup of `homebrew/core` is pretty complete 😄.

> fixed the grotesque lack of spacing in post

👍 🎉

**Comment by ilovezfs** [comment]
>I think at this point the coup of homebrew/core is pretty complete :smile:.

:v:

**Pull request closed without merging**